nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -3561,180 +3589,6 @@ pub mod test {
         logctx.cleanup_successful();
     }
 
-    #[test]
-    fn test_invalid_parent_blueprint_two_zones_with_same_external_ip() {
```

Comment:
I removed these tests because they were sensitive to the change in ordering of moving `ensure_input_networking_records_appear_in_parent_blueprint` from where it appeared in `BuilderExternalNetworking::new()` to just before calling `BuilderExternalNetworking::new()` - we get _different_ errors instead of the specific errors these tests were checking for.

I don't think there's an easy way to fix the tests, because since these tests were written, we've introduced checks on constructing `PlanningInput` itself that [ensures we don't have any duplicate IPs / NIC IPs / MACs](https://github.com/oxidecomputer/omicron/blob/69de8b6288fde36fbcd6cabb4d632d62851230ad/nexus/types/src/deployment/network_resources.rs#L79-L107). Particularly given the checks we're doing there, these seemed worthy of being dropped.